addons/api/mirage/generated/factories/host-catalog.js

```diff
@@ -1,6 +1,22 @@
 import { Factory } from 'ember-cli-mirage';
-import { random, date, datatype } from 'faker';
+import { random, date, datatype, address } from 'faker';
 
+/**
+ *
+ * @param {object} plugin
+ * @returns
+ */
+function isAzurePlugin(plugin) {
+  if (plugin && plugin.name === 'azure') {
+    return true;
+  }
+}
+
+function isAwsPlugin(plugin) {
+  if (plugin && plugin.name === 'aws') {
+    return true;
+  }
+}
 /**
  * GeneratedHostCatalogModelFactory
```

Comment:
This logic is far too bespoke for the generated factory and should live in the regular factory.

Mirage mocks must return `attributes`, since it mimics the API (not the Ember models).

Refer to the account factory as a reference for type-specific mocking.  It uses a switch statement that is fairly readable.